src/nupic/research/frameworks/htm/temporal_memory/pair_memory_apical_tiebreak.py

```diff
@@ -0,0 +1,117 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2022, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from nupic.research.frameworks.htm.temporal_memory import TemporalMemoryApicalTiebreak
+
+import torch
+
+real_type = torch.float32
+int_type = torch.int64
```

Comment:
int64 is a really wide range, do we really use all that.

Reply:
int64 is the minimum width needed for Tensors that index into other Tensors